include/seqan3/search/fm_index/fm_index.hpp

```diff
@@ -223,10 +255,21 @@ class fm_index
      * No guarantees.
      */
     void construct(text_t const & text)
+        //!\cond
+        requires !is_collection
+        //!\endcond
     {
          // text must not be empty
-        if (text.begin() == text.end())
+        if (std::begin(text) == std::end(text))
```

Comment:
```suggestion
        if (std::ranges::begin(text) == std::ranges::end(text))
```

Reply:
What's the reasoning behind this? Is it more general?

Reply:
`std::begin` is a function. The compiler would only look in the begin functions qualified by the std namespace. This means, that if you have an object with a global begin function that is not in std namespace, the correct begin cannot be found. The alternative would be to using declare `using std::begin` in the scope of the function and call it without the namespace qualifier. This would allow for ADL to kick in. The `std::ranges::begin` is a function object and allows for ADL without this using declaration.

Reply:
I think we aliased the `std::ranges::begin` so you can now just use `begin`

Reply:
Well someone should document them.

Reply:
http://docs.seqan.de/seqan/3.0.0-master-user/group__ranges.html#ga8b1352687dc82a2d4922da34778ca215

Reply:
That's the `std::ranges::` aliases. I meant the seqan3 versions.